### PR TITLE
[ODC-5071]: Update web terminal to explain the reason why the workspace was stopped

### PR DESCRIPTION
**Pull request by JPinkney**
This PR updates the web terminal to look into the custom resource annotation ```controller.devfile.io/stopped-by``` to determine why the workspace has stopped. Additionally, when that error is thrown it shows a button that allows you to restart your web terminal.

If WTO 1.1 is installed it just defaults to the old error "The terminal connection has closed" since the `controller.devfile.io/stopped-by` annotation does not exist in the CR.

Before: https://www.youtube.com/watch?v=2S4inkX5sKc
After: https://www.youtube.com/watch?v=YUld4YKFeoU

Signed-off-by: Josh Pinkney <joshpinkney@gmail.com>

### CODE REVIEW

**Comment by serenamarie125** [comment]
Nice usability enhancement for the Web Terminal.  Can we get this reviewed and merged for 4.7 @christianvogt ?

**Comment by serenamarie125** [comment]
FYI, In order for this to be exposed in the Dev Console, WTO 1.2 is needed ( currently planned for a couple weeks after 4.7 GA )

**Comment by sleshchenko** [comment]
>  In order for this to be exposed in the Dev Console, WTO 1.2 is needed ( currently planned for a couple weeks after 4.7 GA )

@JPinkney Could you describe the expected behavior with these changes and WTO 1.1? Thx

**Comment by JPinkney** [comment]
If WTO 1.1 is installed it just defaults to the old error "The terminal connection has closed" since the `controller.devfile.io/stopped-by` annotation does not exist in the CR

**Comment by sleshchenko** [comment]
Tested on crc with custom-built images and works as expected and demoed in the video.

One minor comment that is not really related to the purpose of this PR.
When list devworkspaces or exec/init fails OpenShift Console still shows `try again` link that just refresh the browser page. Maybe it makes sense to rework it with the new Restart terminal button that would just retry init terminal widget?

Screenshot_20201119_164647

It may be better to limit the scope of this PR and do not add the proposed changes even if they make sense.

**Review by sleshchenko — APPROVED** [review]

**Comment by JPinkney** [comment]
> When list devworkspaces or exec/init fails OpenShift Console still shows try again link that just refresh the browser page. Maybe it makes sense to rework it with the new Restart terminal button that would just retry init terminal widget?

Yeah, I think we can go that as an improvement in the future just so we can have this code reviewed and merged asap

File: frontend/packages/console-app/src/components/cloud-shell/CloudShellExec.tsx [hunk 1]
```diff
@@ -142,27 +177,63 @@ const CloudShellExec: React.FC<CloudShellExecProps> = ({
...
+  ]);
 
   if (wsError) {
     return (
-      <LoadError
-        message={wsError}
-        label={t('cloudshell~OpenShift command line terminal')}
-        canRetry={false}
-      />
+      <div className="co-cloudshell-terminal__container-error">
```
Comment (christianvogt) [inline frontend/packages/console-app/src/components/cloud-shell/CloudShellExec.tsx:201]: Create a new `CloudShellExec.scss` file for the styles owned by this component
```suggestion
      <div className="co-cloudshell-exec__container-error">
```

File: frontend/packages/console-app/src/components/cloud-shell/CloudShellExec.tsx [hunk 2]
```diff
@@ -142,27 +177,63 @@ const CloudShellExec: React.FC<CloudShellExecProps> = ({
...
+        <EmptyState>
+          <EmptyStateBody className="cloudshell-error">{wsError}</EmptyStateBody>
```
Comment (christianvogt) [inline frontend/packages/console-app/src/components/cloud-shell/CloudShellExec.tsx:203]:
```suggestion
          <EmptyStateBody className="co-cloudshell-exec__error-msg">{wsError}</EmptyStateBody>
```

File: frontend/packages/console-app/src/components/cloud-shell/CloudShellTerminal.scss [hunk 3]
```diff
@@ -1,7 +1,18 @@
...
   }
+
+  &__container-error {
+    background-color: #fff;
+    height: 100%;
+
+    .cloudshell-error {
+      color: $color-pf-red !important;
+    }
+  }
```
Comment (christianvogt) [inline frontend/packages/console-app/src/components/cloud-shell/CloudShellTerminal.scss:17]: Move to `CloudShellExec.scss`

**Comment by JPinkney** [comment]
@christianvogt I'm a bit confused, the tests are failing with what looks like `\ No newline at end of file` but I've tried both with and without a newline and the tests still fail. Is it possible that `i18n files are not up to date. Commit them to fix.` is causing the error?

**Comment by christianvogt** [comment]
Did you run the `i18n` yarn script? or manually update the file? Try to run it again and commit any change.

**Comment by JPinkney** [comment]
Ah I didn't know I had to run `yarn i18n` that was the issue! thanks!

**Pull request merged**